src/config.cpp

```diff
@@ -59,14 +66,116 @@ void parse_uint32(void* value_ptr, const char* txt) {
     cfgval->set = true;
 }
 
+// Helper function to trim whitespace.
+std::string trim(const std::string& str) {
+    size_t first = str.find_first_not_of(" \t\n");
+    size_t last = str.find_last_not_of(" \t\n");
+
+    // Check for valid range
+    if (first == std::string::npos || last == std::string::npos) {
+        return str; // Empty or only whitespace
+    }
+
+    // Extract the trimmed string
+    std::string trimmed = str.substr(first, (last - first + 1));
+    return trimmed;
+}
+
+std::string get_clvk_env_name(const std::string& name) {
+    std::string var_name = "CLVK_";
+    std::string optname_upper(name);
+    std::transform(optname_upper.begin(), optname_upper.end(),
+                   optname_upper.begin(), ::toupper);
+    var_name += optname_upper;
+    return var_name;
+}
+
+void read_config_file(std::unordered_map<std::string, std::string>& umap,
+                      std::ifstream& config_stream) {
+
+    std::string line;
+    while (std::getline(config_stream, line)) {
+        // Ignore comments and empty lines
+        line = trim(line);
+        if (line.empty() || line[0] == '#') {
+            continue;
+        }
+
+        // Parse key-value pairs
+        size_t pos = line.find('=');
+        if (pos != std::string::npos) {
+            std::string key = trim(line.substr(0, pos));
+            std::string value = trim(line.substr(pos + 1));
+            // Store values (if any)
+            if (value != "") {
+                umap[key] = value;
+                cvk_debug_group_fn(loggroup::cfg, "'%s' = '%s'", key.c_str(),
+                                   value.c_str());
+            }
+        } else {
+            cvk_warn_group_fn(loggroup::cfg, "%s , %s",
+                              "The following line is malformed", line.c_str());
+        }
+    }
+    config_stream.close();
+}
+
+void parse_config_file() {
+    std::unordered_map<std::string, std::string> file_config_values;
+    std::string conf_file = "clvk.conf";
+    std::ifstream config_stream;
+
+    std::vector<std::string> config_file_paths;
+    config_file_paths.push_back("/etc/clvk.conf");
+    config_file_paths.push_back("~/.config/clvk.conf");
+    config_file_paths.push_back(
+        (std::filesystem::current_path() / conf_file).string());
+    // First check if env var has file
+    std::string conv_file_env_var = "CLVK_CONFIG_FILE";
```

Comment:
Why did you introduce a function to compose the name of environment variables used for configuration without using it?